flepimop/gempyor_pkg/tests/shared_cli/test_duration_param_type_class.py

```diff
@@ -0,0 +1,40 @@
+from datetime import timedelta
+from typing import Any
+
+from click.exceptions import BadParameter
+import pytest
+
+from gempyor.shared_cli import DurationParamType
+
+
+@pytest.mark.parametrize("nonnegative", (True, False))
+@pytest.mark.parametrize("value", ("abc", "$12.34", "12..3", "12years", "12.a2"))
```

Comment:
is forbidding "years" about that being conceptually wrong or simply a duration we don't want to contemplate? seems as a generic thing, its appropriate for the tool to potentially support years, so a bit weird to explicitly test that it *doesn't*.

Reply:
Two issues with years: 1) years do not have a constant length (2024 was 366 days but most years are 365 days), and 2) python's `timedelta` class which doesn't have a `years` argument: https://docs.python.org/3/library/datetime.html#datetime.timedelta.

I could however see an argument for adding micro/milliseconds on conceptual ground if you would like? I just excluded those since it was too short for most practical use cases.

Reply:
so this is a duration we explicitly don't want to contemplate (primarily because it's an ambiguous unit for our case) - that's fine! It does feel worth noting somewhere that we don't allow this unit of time - like maybe in the method documentation? But I don't think we need, say, a special error message for years.

re ms - I think it's fine to not support it _at this time_. I think you're saying this would be one that we'd be willing to contemplate (i.e. we aren't testing to ensure its exclusion), but that we haven't bothered to yet support (because we don't think its practically useful).